internal/deviations/deviations.go

```diff
@@ -66,9 +66,37 @@ package deviations
 import (
 	"flag"
 
+	log "github.com/golang/glog"
+	"github.com/openconfig/featureprofiles/internal/metadata"
+	mpb "github.com/openconfig/featureprofiles/proto/metadata_go_proto"
 	"github.com/openconfig/ondatra"
 )
 
+// lookupDutDeviations returns the deviations for the specified dut. If no duts match `nil` is returned.
+func lookupDUTDeviations(dut *ondatra.DUTDevice) *mpb.Metadata_Deviations {
+	for _, platformExceptions := range metadata.Get().PlatformExceptions {
+		vendor := platformExceptions.GetPlatform().Vendor.String()
+		for _, hardwareModel := range platformExceptions.GetPlatform().HardwareModel {
+			if vendor == dut.Device.Vendor().String() && hardwareModel == dut.Device.Model() {
+				return platformExceptions.GetDeviations()
+			}
+		}
+	}
+	log.Warningf("No platform exceptions for dut platform %v or model %v configured in platform exceptions metadata %v", dut.Device.Vendor().String(), dut.Device.Model(), metadata.Get().PlatformExceptions)
+	return nil
+}
+
+func logErrorIfFlagSet(name string) {
+	if !flag.Parsed() {
```

Comment:
I think you can assume flags have been parsed by this point and remove this if statement. There is no way to reach this point without flag.Parse being called.

Reply:
done